msal/authority.py

```diff
@@ -37,18 +42,30 @@ def __init__(self, authority_url, validate_authority=True,
         self.verify = verify
         self.proxies = proxies
         self.timeout = timeout
-        canonicalized, self.instance, tenant = canonicalize(authority_url)
-        tenant_discovery_endpoint = (
-            'https://{}/{}{}/.well-known/openid-configuration'.format(
-                self.instance,
-                tenant,
-                "" if tenant == "adfs" else "/v2.0" # the AAD v2 endpoint
-                ))
-        if (tenant != "adfs" and validate_authority
+        authority, self.instance, tenant = canonicalize(authority_url)
+        is_b2c = self.instance.endswith(".b2clogin.com")
```

Comment:
only issue here is that we also have to support b2c in the sovereign clouds...so these also need to work:
 *.b2clogin.us, *.b2clogin.cn, and *.b2clogin.de

That's why MSAL.NET is stuck on the /tfp/ validation to determine if this is a b2c authority or not. if b2c has instance discovery, like ests, this would not be needed.

Reply:
Good point. I did not know B2C sovereign hosts before. Now I've added them by the latest commit, so it would also work for the sovereign clouds you just mentioned. Hello world, happy B2C! :-)